python-sdk/src/astro/files/types/ndjson.py

```diff
@@ -76,14 +76,18 @@ def flatten(
             else:
                 extra_rows = []
 
-            df = pd.DataFrame(
-                pd.json_normalize([json.loads(row) for row in rows], **normalize_config)
+            # ToDo: Check if there is an alternative to pandas json_normalize(). A custom implementation is much faster.
+            # But that would entail breaking changes.
+            # ref : https://stackoverflow.com/questions/69945941/
+            #   why-does-pandas-json-normalize-run-slower-than-my-hand-crafted-loop
+            df = pd.json_normalize(
+                [json.loads(row) for row in rows], **normalize_config
             )
-            if result_df is None:
-                result_df = df
-            else:
-                result_df = pd.concat([result_df, df])
+            result_df.append(df)
 
-            row_count = result_df.shape[0]
+            row_count = row_count + df.shape[0]
 
-        return result_df
+        # Running concat multiple times is much slower instead we are appending all the generated dataframes
+        # in a list and then concatenating in single call. This brought down the cost from 351.79550790786743
```

Comment:
what's 351? Is it seconds?

Reply:
I'm also interested, is this a significant time reduction?

Reply:
@kaxil  Yes, It's seconds.

Reply:
@dimberman Earlier the data was never getting loaded in the table, because of that, we have seen a performance boost of 25sec for one GB. Now that we are loading data in the table, we have this new benchmark.

Reply:
@dimberman Do you have any suggestions to improve the performance further?